File: wagerproof-ios-native/WagerproofKit/Sources/WagerproofSharedKit/AppGroup.swift
import Foundation

public enum AppGroup {
    public static let identifier = "group.com.wagerproof.mobile"

    public static var defaults: UserDefaults {
        guard let d = UserDefaults(suiteName: identifier) else {
            // Fall back to standard defaults if the app group isn't configured —
            // this lets unit tests run without the entitlement. In production
            // the entitlement is always present (declared in Wagerproof.entitlements).
            return .standard
        }
        return d
    }
}

public enum AppGroupKey {
    public static let lastNotificationRoute = "last_notification_route"
    public static let themePreference = "theme_pref"
    public static let adminModeEnabled = "admin_mode_enabled"
    /// Kept byte-for-byte compatible with the original RN widget bridge.
    /// Both the app and widget extension must use this exact key.
    public static let widgetPayload = "widgetPayload"
    /// DEBUG-only: when true, stores serve bundled real-data fixtures instead
    /// of hitting Supabase (offseason UI development). See `DummyDataMode`.
    public static let dummyDataMode = "dummy_data_mode_debug"
    // B08: coarse-grained subscription snapshot consumed by widgets + cold
    // launch UI so they don't flash a "free" state while RevenueCat reconciles
    // on the network. Source of truth remains RevenueCat; this is a mirror.
    public static let proEntitlementGranted = "pro_entitlement_granted_v1"
    public static let proSubscriptionType = "pro_subscription_type_v1"
    /// DEBUG-only: selected WagerBot chat model id (see `WagerBotModelSelection`).
    /// Non-default values route the chat to the parallel `wagerbot-agent` function.
    public static let wagerBotChatModel = "wagerbot_chat_model_debug"

    // Picks-expiry hold (`PicksExpiryService`). Persisted rather than held in
    // memory so the paywall pill and the Live Activity agree on one deadline
    // across relaunches — a countdown that restarts on cold launch reads as
    // fake and torches the urgency it exists to create.
    public static let picksExpiryStartedAt = "picks_expiry_started_at_v1"
    public static let picksExpiryPickCount = "picks_expiry_pick_count_v1"
    public static let picksExpiryAgentName = "picks_expiry_agent_name_v1"

    /// Per-user onboarding completion key. Matches RN's
    /// `@wagerproof/onboarding-completed/{userId}` AsyncStorage key. The
    /// previous global `onboarding_complete` key leaked completion state
    /// across user switches on shared devices, AND a fresh install on a
    /// new device with an already-onboarded profile would replay the flow
    /// because no Supabase round-trip was reading the server flag.
    public static func onboardingComplete(userId: String) -> String {
        "onboarding_complete/\(userId)"
    }
}

File: wagerproof-android-native/core/shared/src/main/kotlin/com/wagerproof/core/shared/AppGroup.kt
package com.wagerproof.core.shared

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences

/**
 * Android analog of iOS's App Group UserDefaults suite (WagerproofSharedKit/AppGroup.swift).
 * One SharedPreferences file shared between the app and Glance widgets (same process on
 * Android, so a plain prefs file is sufficient — no cross-process container needed).
 *
 * Call [initialize] from Application.onCreate before any service touches [prefs].
 */
object AppGroup {
    /** Kept identical to the iOS App Group identifier so payload tooling greps match. */
    const val IDENTIFIER = "group.com.wagerproof.mobile"

    @SuppressLint("StaticFieldLeak") // application context only — never an Activity
    @Volatile
    private var appContext: Context? = null

    fun initialize(context: Context) {
        appContext = context.applicationContext
    }

    /** Application context for services that need one lazily (widgets, notifications). */
    val context: Context
        get() = requireNotNull(appContext) {
            "AppGroup.initialize(context) must be called from Application.onCreate"
        }

    val prefs: SharedPreferences
        get() = context.getSharedPreferences(IDENTIFIER, Context.MODE_PRIVATE)
}

/** Storage keys — byte-for-byte with iOS `AppGroupKey` (and RN where noted). */
object AppGroupKey {
    const val LAST_NOTIFICATION_ROUTE = "last_notification_route"
    const val THEME_PREFERENCE = "theme_pref"
    const val ADMIN_MODE_ENABLED = "admin_mode_enabled"

    /**
     * Declared for parity with iOS, but NOTE: the live widget services use the
     * legacy literal `"widgetPayload"` (see [WIDGET_PAYLOAD_LEGACY]) for
     * Expo-widget compat. Do not switch widgets to this key.
     */
    const val WIDGET_PAYLOAD = "widget_payload_v1"

    /** The key the widget payload is actually stored under (legacy Expo-compat). */
    const val WIDGET_PAYLOAD_LEGACY = "widgetPayload"

    /** DEBUG-only: stores serve bundled fixtures instead of Supabase. */
    const val DUMMY_DATA_MODE = "dummy_data_mode_debug"

    // Coarse subscription snapshot mirrored for widgets/cold-launch so the UI
    // doesn't flash "free" while RevenueCat reconciles. RevenueCat stays the
    // source of truth.
    const val PRO_ENTITLEMENT_GRANTED = "pro_entitlement_granted_v1"
    const val PRO_SUBSCRIPTION_TYPE = "pro_subscription_type_v1"

    /** DEBUG-only WagerBot chat model picker (see WagerBotModelSelection). */
    const val WAGERBOT_CHAT_MODEL = "wagerbot_chat_model_debug"

    /** Per-user onboarding completion — matches RN `@wagerproof/onboarding-completed/{userId}` semantics. */
    fun onboardingComplete(userId: String): String = "onboarding_complete/$userId"
}
